corehq/apps/hqwebapp/static/hqwebapp/js/bootstrap3/email-request.js

```diff
@@ -0,0 +1,132 @@
+
+        self.isRequestReportSubmitting = false;
+
+        self.reportUrl = ko.observable('');
+
+        self.$formElement.submit(() => {
+            resetErrors();
+
+            const isDescriptionEmpty = !self.subjectText() && !self.descriptionText();
+            if (isDescriptionEmpty) {
+                self.hasSubjectError(true);
+            }
+
+            if (self.emailInput() && !isValidEmail(self.emailInput())) {
+                self.hasEmailInputError(true);
+                return false;
+            }
+
+            const emailAddresses = self.otherEmailsText().replace(/ /g, "").split(",");
+            for (const email of emailAddresses) {
+                if (email && !isValidEmail(email)) {
+                    self.hasRecipientsInputError(true);
+                    return false;
+                }
+            }
+            if (isDescriptionEmpty) {
+                return false;
+            }
+
+            if (!self.isRequestReportSubmitting && self.$submitBtn.text() === self.$submitBtn.data("success-text")) {
+                self.$element.modal('hide');
+            } else if (!self.isRequestReportSubmitting) {
+                self.$submitBtn.button('loading');
+                self.cancelBtnEnabled(false);
+                self.$formElement.ajaxSubmit({
+                    type: "POST",
+                    url: self.$formElement.attr('action'),
+                    beforeSerialize: hqwebappRequestReportBeforeSerialize,
+                    beforeSubmit: hqwebappRequestReportBeforeSubmit,
+                    success: hqwebappRequestReportSucccess,
+                    error: hqwebappRequestReportError,
+                });
+            }
+            return false;
+        });
+
+        self.openModal = function () {
+            self.subjectHasFocus(true);
+        };
+
+        self.resetForm = function () {
+            self.$formElement.find("button[type='submit']").button('reset');
+            self.$formElement.resetForm();
+            self.cancelBtnEnabled(true);
+            self.$submitBtn.button('reset');
+            self.hasEmailInputError(false);
+        };
+
+        function isValidEmail(email) {
+            var regex = /^([a-zA-Z0-9_.+-])+@(([a-zA-Z0-9-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+            return regex.test(email);
+        }
+
+        function resetErrors() {
+            self.hasSubmitError(false);
+            self.hasSubjectError(false);
+            self.hasEmailInputError(false);
+            self.hasOtherEmailInputError(false);
+            self.hasRecipientsInputError(false);
+        }
+
+        function hqwebappRequestReportBeforeSerialize() {
+            self.reportUrl(location.href);
+        }
+
+        function hqwebappRequestReportBeforeSubmit() {
+            self.isRequestReportSubmitting = true;
+        }
+
+        function hqwebappRequestReportSucccess() {
+            self.isRequestReportSubmitting = false;
+            self.$submitBtn.button('success');
```

Comment:
what is "success" here?

Reply:
This will put the form submit button in the success state. Specifically, it will cause the text from the `data-success-text` binding to be shown.